src/python/pants/backend/python/util_rules/interpreter_constraints.py

```diff
@@ -154,6 +160,9 @@ def create_from_compatibility_fields(
         constraint_sets = {field.value_or_global_default(python_setup) for field in fields}
         # This will OR within each field and AND across fields.
         merged_constraints = cls.merge_constraint_sets(constraint_sets)
+        if not merged_constraints:
```

Comment:
This makes sense because it's reasonable for a user to want to work against just python_requirement targets, say. And this behavior is "continuous" in the sense that things get less constrained the fewer targets you have adding constraints.